op-program/chainconfig/chaincfg.go

```diff
@@ -13,16 +16,52 @@ func OPSepoliaChainConfig() *params.ChainConfig {
 	return mustLoadChainConfig("op-sepolia")
 }
 
+//go:embed configs/*json
+var customChainConfigFS embed.FS
+
 func RollupConfigByChainID(chainID uint64) (*rollup.Config, error) {
 	config, err := rollup.LoadOPStackRollupConfig(chainID)
+	if err == nil {
+		return config, err
+	}
+	return rollupConfigByChainID(chainID, customChainConfigFS)
+}
+
+func rollupConfigByChainID(chainID uint64, customChainFS embed.FS) (*rollup.Config, error) {
+	// Load custom rollup configs from embed FS
+	file, err := customChainFS.Open(fmt.Sprintf("configs/%d-rollup.json", chainID))
 	if err != nil {
 		return nil, fmt.Errorf("failed to get rollup config for chain ID %d: %w", chainID, err)
 	}
-	return config, nil
+	dec := json.NewDecoder(file)
+	dec.DisallowUnknownFields()
+	var customRollupConfig rollup.Config
+	if err := dec.Decode(&customRollupConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse rollup config for chain ID %d: %w", chainID, err)
+	}
+	return &customRollupConfig, nil
 }
 
 func ChainConfigByChainID(chainID uint64) (*params.ChainConfig, error) {
-	return params.LoadOPStackChainConfig(chainID)
+	config, err := params.LoadOPStackChainConfig(chainID)
+	if err == nil {
+		return config, err
+	}
+	return chainConfigByChainID(chainID, customChainConfigFS)
+}
+
+func chainConfigByChainID(chainID uint64, customChainFS embed.FS) (*params.ChainConfig, error) {
+	// Load from custom chain configs from embed FS
+	data, err := customChainFS.ReadFile(fmt.Sprintf("configs/%d-genesis-l2.json", chainID))
+	if err != nil {
```

Comment:
Similar to above, probably worth having a specific error message for os.ErrNotExist.